src/java.base/share/classes/java/util/regex/Pattern.java

```diff
@@ -5091,11 +5097,18 @@ boolean match(Matcher matcher, int i, CharSequence seq) {
                             return false;
                     }
                 }
-                x += Character.charCount(c1);
+                int xIncr = Character.charCount(c1);
+                x += xIncr;
                 j += Character.charCount(c2);
+
+                if(xIncr > 1) {
```

Comment:
You can eliminate `xIncr` by comparing `c1 >= Character.MIN_SUPPLEMENTARY_CODE_POINT` here.

Reply:
Nice! Thanks will do.